clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -617,7 +618,7 @@ public boolean commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets,
 
             now = time.milliseconds();
             remainingMs = timeoutMs - (now - startMs);
-        } while (remainingMs > 0);
+        } while (remainingMs > 0 && !isFutureCompleted);
```

Comment:
Thanks for pointing the correct fix and review comments @ijuma. Sure, I will update the PR soon after adding the test.